apps/site/assets/ts/schedule/components/ScheduleFinder.tsx

```diff
@@ -36,9 +37,51 @@ const ScheduleFinder = ({
   const [selectedDirection, setSelectedDirection] = useState(directionId);
   const [selectedOrigin, setSelectedOrigin] = useState<SelectedOrigin>(null);
 
+  const updateURL = (
+    directionId: DirectionId,
+    origin: SelectedOrigin
+  ): void => {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    const query = { direction_id: directionId.toString(), origin: origin };
+    const newLoc = updateInLocation(query, window.location);
+    // newLoc is not a true Location, so toString doesn't work
+    window.history.replaceState({}, "", `${newLoc.pathname}${newLoc.search}`);
+  };
+
   const changeDirection = (direction: DirectionId): void => {
     setSelectedOrigin(null);
     setSelectedDirection(direction);
+    updateURL(direction, null);
+  };
+
+  const changeOrigin = (origin: SelectedOrigin): void => {
+    setSelectedOrigin(origin);
+    if (selectedDirection !== null && origin !== null) {
+      updateURL(selectedDirection, origin);
+    }
+  };
+
+  const getCurrentScheduleFinderValues = (): void => {
+    let params = window.location.search;
+
+    if (params.startsWith("?")) {
```

Comment:
@alicia-mbta since we're using `use-query-params` we can and should use its capabilities for parsing params, rather than rolling our own. See https://github.com/pbeshai/use-query-params.